File: Offer/2285/main.go
package main

func isStraight(nums []int) bool {
	hashmap := make(map[int]bool)
	var ok bool
	minValue, maxValue := 13, 1
	for _, num := range nums {
		if _, ok = hashmap[num]; ok {
			return false
		}
		if num != 0 {
			hashmap[num] = true
			if num < minValue {
				minValue = num
			}
			if num > maxValue {
				maxValue = num
			}
		}
	}
	return maxValue-minValue < 5
}

File: Offer/2285/main.py
from typing import Dict, List


class Solution:
    def isStraight(self, nums: List[int]) -> bool:
        hashmap: Dict[int, bool] = {}
        minValue, maxValue = 13, 1
        for num in nums:
            if hashmap.get(num) != None:
                return False
            if num != 0:
                hashmap[num] = True
                if num < minValue:
                    minValue = num
                if num > maxValue:
                    maxValue = num
        return maxValue - minValue < 5